### PR TITLE
Add ToolPart localization

### PR DESCRIPTION
**Pull request by albus12138**
Add ToolPart localization.

### CODE REVIEW

File: build.properties [hunk 1]
```diff
@@ -1,6 +1,11 @@
...
-tgregworks.version=1.0.18
+tgregworks.version=1.1.0
 
-gregtech.version=gregtech_1.7.10:5.07.07
+tictooltips.version=1.2.5
+cofhcore.version=[1.7.10]3.1.4-329
+mantle.version=0.3.2.jenkins187
+iguanatinkertweaks.version=2.1.7.4
+tconstruct.version=1.8.4.build951
+gregtech.version=5.09.33.52
```
Comment (Vexatos) [inline build.properties:11]: Wrong GT version. This has to build against 5.07.07.

File: src/main/java/vexatos/tgregworks/item/ItemTGregPart.java [hunk 2]
```diff
@@ -45,18 +46,16 @@ public String getItemStackDisplayName(ItemStack stack) {
...
 		} else {
-			matName = Materials.get(data.getString("material")).mDefaultLocalName;
+			matName = Materials.get(data.getString("material")).mLocalizedName;
 		}
 
-		matName = matName + " " + type.getPartName();
-		//String material = StatCollector.translateToLocal("tgregworks.parttype." + matName);
-		//String name = StatCollector.translateToLocal("tgregworks.toolpart." + PartTypes.getFromID(stack.getItemDamage()) + "." + matName);
-		//name = name.replaceAll("%%material", material);
+		String name = StatCollector.translateToLocal("tgregworks.toolpart." + type.getID());
```
Comment (Vexatos) [inline src/main/java/vexatos/tgregworks/item/ItemTGregPart.java:54]: Instead of the ID, one could use something like `type.name()`. Would probably be more sane in the lang file.

File: build.gradle [hunk 3]
```diff
@@ -28,14 +28,30 @@ group= "vexatos.tgregworks" // http://maven.apache.org/guides/mini/guide-naming-
...
+    }
+
+    ivy {
+        name = 'gtnh_download_source_stupid_underscore_typo'
+        artifactPattern("http://downloads.gtnewhorizons.com/Mods_for_Jenkins/[module]_[revision].[ext]")
+    }
+
+    ivy { // GregTech
+        name = 'gtnh_download_source'
+        artifactPattern("http://downloads.gtnewhorizons.com/Mods_for_Jenkins/[module]-[revision].[ext]")
```
Comment (Vexatos) [inline build.gradle:48]: I would prefer to depend on official GregTech sources; modifying the build files should not be part of this PR.

**Review by Vexatos — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (albus12138) [reply build.gradle:48]: In fact, I cannot build with the original build.gradle (lots of dependencies cannot be found), so I use dependencies from GTNH's repo. Same reason for the change of GT version.
If you are sure you can build with the original build.gradle, I will reset this file.

File: src/main/java/vexatos/tgregworks/item/ItemTGregPart.java (hunk 2 above)
Comment (albus12138) [reply src/main/java/vexatos/tgregworks/item/ItemTGregPart.java:54]: Ok, I will fix this in next commit.

File: src/main/java/vexatos/tgregworks/item/ItemTGregPart.java [hunk 4]
```diff
@@ -45,18 +46,16 @@ public String getItemStackDisplayName(ItemStack stack) {
...
 		} else {
-			matName = Materials.get(data.getString("material")).mDefaultLocalName;
+			matName = Materials.get(data.getString("material")).mLocalizedName;
 		}
 
-		matName = matName + " " + type.getPartName();
-		//String material = StatCollector.translateToLocal("tgregworks.parttype." + matName);
-		//String name = StatCollector.translateToLocal("tgregworks.toolpart." + PartTypes.getFromID(stack.getItemDamage()) + "." + matName);
-		//name = name.replaceAll("%%material", material);
+		String name = StatCollector.translateToLocal("tgregworks.toolpart." + type.getPartName().replace(" ", "_"));
```
Comment (Vexatos) [inline src/main/java/vexatos/tgregworks/item/ItemTGregPart.java:54]: Would it be possible to use `type.name().toLowerCase()` Instead? That's what I tried suggesting.

File: src/main/java/vexatos/tgregworks/item/ItemTGregPart.java [hunk 5]
```diff
@@ -45,18 +46,16 @@ public String getItemStackDisplayName(ItemStack stack) {
...
+		matName = name.replaceAll("%%material", matName);
```
Comment (Vexatos) [inline src/main/java/vexatos/tgregworks/item/ItemTGregPart.java:55]: Usually, you'd probably use a single percent sign here.

**Review by Vexatos — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (Vexatos) [reply build.gradle:48]: TGw should stay compatible, and therefore build, with the last official GregTech version, at least for now. Regardless, updating the build instructions has nothing to do with the localization options proposed in this PR.

File: src/main/java/vexatos/tgregworks/item/ItemTGregPart.java (hunk 5 above)
Comment (albus12138) [reply src/main/java/vexatos/tgregworks/item/ItemTGregPart.java:55]: Gregtech's source code uses double percent sign, so I use the same style here.

File: src/main/java/vexatos/tgregworks/item/ItemTGregPart.java (hunk 4 above)
Comment (albus12138) [reply src/main/java/vexatos/tgregworks/item/ItemTGregPart.java:54]: I think part name with underline sign is easier for reading, and lowercase is acceptable.

File: build.gradle (hunk 3 above)
Comment (albus12138) [reply build.gradle:48]: Ok, I will rollback these two files. But I hope you could update the unavailable dependencies, in a compatible way. A useless but compatible gradle file is unreasonable.

File: src/main/java/vexatos/tgregworks/integration/TGregRegistry.java [hunk 6]
```diff
@@ -78,7 +78,7 @@ public void registerToolParts() {
 		for(Materials m : toolMaterials) {
 			toolMaterialNames.add(m.mDefaultLocalName);
 			int matID = getMaterialID(m);
-			TConstructRegistry.addToolMaterial(matID, m.name(), m.mDefaultLocalName, m.mToolQuality,
+			TConstructRegistry.addToolMaterial(matID, m.name(), m.mLocalizedName, m.mToolQuality,
```
Comment (Vexatos) [inline src/main/java/vexatos/tgregworks/integration/TGregRegistry.java:81]: This field does not appear to exist on GregTech 5.07.07, so it won't build like this.

**Review by Vexatos — CHANGES_REQUESTED** [review]

File: src/main/java/vexatos/tgregworks/integration/TGregRegistry.java (hunk 6 above)
Comment (albus12138) [reply src/main/java/vexatos/tgregworks/integration/TGregRegistry.java:81]: mDefaultLocalName will only return english name, instead of localized name.

Comment (albus12138) [reply src/main/java/vexatos/tgregworks/integration/TGregRegistry.java:81]: mDefaultLocalName will only return english name, instead of localized name.

**Comment by Vexatos** [comment]
Merged this on a new branch https://github.com/Vexatos/TinkersGregworks/tree/GT-NH for the GTNH pack, since this PR isn't possible on an official GregTech version.
